### PR TITLE
[PromptFlow][Extension] Fixed local batchrun could not be cancelled

### PR DESCRIPTION
**Pull request by Hhhilulu**
# Description
Fixed local batch-run could not be cancelled.

Windows:
image

Linux:
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by guming-llc — DISMISSED** [review]

File: src/promptflow/promptflow/executor/_line_execution_process_pool.py [hunk 1]
```diff
@@ -31,6 +34,29 @@
...
+                logger.warning("Process %s already terminated and not found, skipping", process.pid)
+        logger.info("Successfully terminated process with pid %s", process.pid)
+        process.terminate()
+    except Exception:
+        logger.warning("Error when handling execution stop signal", exc_info=True)
+    finally:
+        sys.exit(1)
+
+
+signal.signal(signal.SIGINT, signal_handler)
```
Comment (thy09) [inline src/promptflow/promptflow/executor/_line_execution_process_pool.py:57]: Can we do this inside some function? It is not a good practice to do stuffs in the module level.

File: src/promptflow/promptflow/executor/_line_execution_process_pool.py [hunk 2]
```diff
@@ -348,13 +374,21 @@ def run(self, batch_inputs):
...
                     self._timeout_process_wrapper,
                     [
                         (run_start_time, self._inputs_queue, self._line_timeout_sec, result_list)
                         for _ in range(self._n_process)
                     ],
                 )
+                try:
+                    while not async_result.ready():
+                        async_result.wait(1)
+                except KeyboardInterrupt:
```
Comment (thy09) [inline src/promptflow/promptflow/executor/_line_execution_process_pool.py:390]: Add some logs here?

Comment (Hhhilulu) [reply src/promptflow/promptflow/executor/_line_execution_process_pool.py:382]: Added

File: src/promptflow/promptflow/executor/_line_execution_process_pool.py (hunk 1 above)
Comment (Hhhilulu) [reply src/promptflow/promptflow/executor/_line_execution_process_pool.py]: Add `signal.signal(signal.SIGINT, signal_handler)` to function level. 
Thank you for your advice.

**Review by guming-llc — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**